pandas/core/indexes/accessors.py

```diff
@@ -126,6 +126,43 @@ class DatetimeProperties(Properties):
     """
 
     def to_pydatetime(self):
+        """
+        Return an ndarray of native Python datetime objects.
+
+        Timezone information is retained if present.
+
+        Returns
+        -------
+        numpy.ndarray
+            object dtype array containing native Python datetime objects.
+
+        Examples
+        --------
+        This method is available on both Series with datetime values, under the
+        ``.dt`` accessor, and directly on DatetimeIndex.
+
+        **Series**
+
+        >>> s = pd.Series(pd.date_range('20180310', periods=2))
+        >>> s.head()
+        0   2018-03-10
+        1   2018-03-11
+        dtype: datetime64[ns]
+
+        >>> s.dt.to_pydatetime()
+        array([datetime.datetime(2018, 3, 10, 0, 0),
+               datetime.datetime(2018, 3, 11, 0, 0)], dtype=object)
+
+        **DatetimeIndex**
+        >>> idx = pd.date_range("2018-03-10", periods=2)
```

Comment:
I'd leave a blank line between the title and the code, as you did with Series.

Reply:
I actually don't think we need a DatetimeIndex example? As this are the docs specific to `Series.dt.to_pydatetime` (they don't share their docstring)

Reply:
Ah, my apologies. I didn't realize they don't share.

Reply:
Opened https://github.com/pandas-dev/pandas/issues/20306 for it